### PR TITLE
Team "boxes"

### PR DESCRIPTION
**Pull request by Joshowens16**
- Adds the basic UI for team boxes/sides.
- If the conditions are met in the functions, the player is added to their respective team and role.
- A player cannot change roles or teams once they've picked, this can easily be changed in the future but it's a bit easier for testing in my opinion. 
- When a player joins a team, it pushes to redux and triggers a rerender for all players.

### CODE REVIEW

File: src/components/RoomView.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { useDispatch, useSelector } from "react-redux";
+import { setRoomId } from "../store/playerSlice";
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import { onValue, ref, set } from "firebase/database";
+import { database } from "../utils/firebase";
+import { setAllPlayers } from "../store/allPlayersSlice";
+import RedTeamBox from "./teamBoxes/TeamOneBox";
+import BlueTeamBox from "./teamBoxes/TeamTwoBox";
+const RoomView = () => {
```
Comment (ocjarman) [inline src/components/RoomView.jsx:11]: i think this file might need to be deleted. this is the roomview before the refactor and might cause some issues in main

File: src/components/teamBoxes/TeamOneBox.jsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  const roomId = useSelector((state) => state.player.roomId);
+  const username = useSelector((state) => state.player.username);
+  const dispatch = useDispatch();
+  playerRef = ref(database, "players/" + playerId);
+  const teamOneRef = ref(database, `rooms/${roomId}/team-1/`);
+  const teamTwoRef = ref(database, `rooms/${roomId}/team-2/`);
+  const teamOneOperativesRef = ref(database, `rooms/${roomId}/team-1/operatives/`);
+  const teamOneSpymasterRef = ref(database, `rooms/${roomId}/team-1/spymaster/`);
+  const { teamOneOperatives } = useSelector(state => state.teamOne);
+  const { teamOneSpymaster } = useSelector(state => state.teamOne);
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamOneBox.jsx:20]: these be combined in one line, right? I usually write my useSelectors a bit differently so i'm not sure!

File: src/components/teamBoxes/TeamOneBox.jsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+  
+  // On click event for a player to be able to join team-1 team as a operative
+  const joinTeamOneOp = async () => {
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamOneBox.jsx:24]: this logic seems like it took a lot of thought! nice job

File: src/components/teamBoxes/TeamOneBox.jsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    <div className="redBoxCard">
+      <div>Team 1</div>
+      <div className="redOpsAndSpys">
+        <div>
+          <p>Operative(s)</p>
+          {
+            teamOneOperatives.map((player, index)=> {
+              return <p
+              key={index}
+              >{player.username}</p>
```
Comment (ocjarman) [inline src/components/teamBoxes/TeamOneBox.jsx:139]: maybe at some point we can break players out into their own component but for now this looks great

File: src/components/teamBoxes/TeamOneBox.jsx (hunk 2 above)
Comment (Joshowens16) [reply src/components/teamBoxes/TeamOneBox.jsx:20]: Not sure where I picked this up but its just deconstructing the "teamOneOperatives" out of state.teamOne! So its the same as "const teamOneOperatives = useSelector(state => state.teamOne.teamOneOperatives)

Comment (ocjarman) [reply src/components/teamBoxes/TeamOneBox.jsx:20]: ohhhh i see. gotcha makes sense.

**Review by ocjarman — APPROVED** [review]
nice work!

**Pull request merged**